### DISCUSSION TITLE
Specifying compression effort

### DISCUSSION THREAD

**Discussion by avidseeker** [ImageMagick/ImageMagick#7869] [Help]
### ImageMagick version

7.1.1.43

### Operating system, version and so on

GNU/Linux

### Description

What effort does Imagemagick use when re-encoding images?

Here's an interesting result I got.

```
$ cjxl -e 9 -q 50 inp.png out.jxl  # size 46K
``` 

However, 

```
$ magick -quality 50 inp.png out.jxl # size 26K
```

Granted, `magick` and `cjxl` must have different concepts of what quality scale is, but while trying to eliminate different factors, I couldn't find an equivalent `-e` option for magick.

What effort profile is it using?

**Comment by snibgo** [comment]
I suppose this is set by "`-define jxl:effort=3`" or whatever. See https://imagemagick.org/script/defines.php :
```
jxl:effort=value 	Set the jpeg-xl encoding effort. Valid values are in the range of 3 (falcon) to 9 (tortoise).
```

**Comment by fmw42** [comment]
Put -quality after reading the input and before the output
